segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java

```diff
@@ -91,9 +91,16 @@ private static Map.Entry<Long, Long> applySearch(StreamSegmentStore store, Strin
             switch (firstElement.getType()) {
                 case Cache: // fallthrough
                 case Storage:
-                    BufferView content = firstElement.getContent().join();
-                    IndexEntry entry = IndexEntry.fromBytes(content);
-                    return entry.getOffset();
+                    firstElement.requestContent(TIMEOUT);
+                    AtomicReference<IndexEntry> entry = new AtomicReference<>();
+                    firstElement.getContent().thenAccept(content -> {
+                        entry.set(IndexEntry.fromBytes(content));
+                    });
+                    if (entry.get() != null) {
+                        return entry.get().getOffset();
```

Comment:
This is a race condition. It will only pass tests if run in an inline executor. In normal execution line 99 will always return null because the .getContent() method will take time to run.

Reply:
Removed the race condition.